src/bar.rs

```diff
@@ -194,36 +194,60 @@ impl Bar {
         let cairo_ctx = cairo::Context::new(&cairo_surf).expect("cairo context");
         cairo_ctx.scale(scale_f, scale_f);
 
-        if !ss.config.blend {
+        if !sfo!(ss, &self.output, blend) {
```

Comment:
I would suggest selecting the palette once and then referring to it directly:

```rust
let palette = match ss.wm_info_provider.is_output_focused(&self.output) {
    true => &ss.config.theme.palette,
    false => &ss.config.theme.unfocused_output,
};
// Then use `palette.thing` instead of `sfo!(ss, &self.output, thing)`.
```

This avoids excessive macro use and redundant calls to `is_output_focused`.